daml-assistant/src/DAML/Assistant/Env.hs

```diff
@@ -181,3 +190,41 @@ getDispatchEnv Env{..} = do
            , (sdkVersionEnvVar, maybe "" versionToString envSdkVersion)
            , (damlAssistantEnvVar, unwrapDamlAssistantPath envDamlAssistantPath)
            ] ++ filter ((`notElem` damlEnvVars) . fst) originalEnv
+
+
+-- | Auto-installs requested version if it is missing and updates daml-assistant
+-- if it is the latest stable version.
+autoInstall :: DamlPath -> Maybe SdkVersion -> IO (Maybe SdkPath)
+autoInstall damlPath sdkVersionM = do
+    damlConfigE <- try $ readDamlConfig damlPath
+    let doAutoInstallME = queryDamlConfig ["auto-install"] =<< damlConfigE
+        doAutoInstallM  = either (const (Just True)) id doAutoInstallME
+        doAutoInstall   = fromMaybe True doAutoInstallM
+    whenMaybe (doAutoInstall && isJust sdkVersionM) $ do
+        let sdkVersion = fromJust sdkVersionM
+            sdkPath = defaultSdkPath damlPath sdkVersion
+
+        unlessM (doesDirectoryExist (unwrapSdkPath sdkPath)) $ do
+            -- sdk is missing, so let's install it!
+            -- first, determine if it is the latest stable version
+            latestVersionE :: Either AssistantError SdkVersion
+                <- try getLatestVersion
```

Comment:
I think this should just be if its newer than what is currently installed

Reply:
Done!